src/test.js

```diff
@@ -280,6 +280,20 @@ Test.prototype = {
 		} );
 
 		if ( module.testsRun === numberOfTests( module ) ) {
+			declareModuleEnd( module );
+
+			// Check if the parent modules, recursively, are done. If that the case,
+			// we emit the `suiteEnd` event and trigger `moduleDone` callback.
+			let parent = module.parentModule;
+			while ( parent && parent.testsRun === numberOfTests( parent ) ) {
+				declareModuleEnd( parent );
+				parent = parent.parentModule;
+			}
+		}
+
+		config.current = undefined;
+
+		function declareModuleEnd( module ) {
```

Comment:
Can we rename this `logSuiteEnd`? It will be similar to `logAssertion` which exists elsewhere, and prefers the js-reporters name (which is what we're moving towards).